tests/kafkatest/services/kafka/kafka.py

```diff
@@ -87,8 +126,11 @@ def start_node(self, node):
     def pids(self, node):
         """Return process ids associated with running processes on the given node."""
         try:
-            return [pid for pid in node.account.ssh_capture("cat /mnt/kafka.pid", callback=int)]
-        except:
+            cmd = "ps ax | grep -i %s | grep java | grep -v grep | awk '{print $1}'" % kafka_dir(node)
```

Comment:
Using the versioned directory here seems a bit risky -- it's almost too restrictive. I know you haven't done it here, but it seems like it'd be easy for users to accidentally do something like

```
node.version = NEW_VERSION
node.restart()
```

Any reason to restrict it like this instead of just detecting any Kafka process?